src/frontend/web_application/src/scenes/MessageList/index.js

```diff
@@ -3,23 +3,24 @@ import { bindActionCreators, compose } from 'redux';
 import { connect } from 'react-redux';
 import { withTranslator } from '@gandi/react-translate';
 import { requestMessages, updateMessage } from '../../store/modules/message';
+import { requestDiscussion } from '../../store/modules/discussion';
 import Presenter from './presenter';
 
-const messagesSelector = state => state.message.messages;
 const messageByIdSelector = state => state.message.messagesById;
 const discussionIdSelector = (state, ownProps) => ownProps.params.discussionId;
 
 const mapStateToProps = createSelector(
-  [messagesSelector, messageByIdSelector, discussionIdSelector],
-  (messages, messagesById, discussionId) => ({
+  [messageByIdSelector, discussionIdSelector],
+  (messagesById, discussionId) => ({
     discussionId,
-    messages: messages
-    .map(messageId => messagesById[messageId])
-    .filter(message => message.discussion_id === discussionId),
+    messages: Object.keys(messagesById)
+      .map(messageId => messagesById[messageId])
```

Comment:
I don't know what you mean by "buggy", though ES2018 Draft, it's already available in core-js/babel